vllm/v1/core/kv_cache_utils.py

```diff
@@ -152,7 +152,7 @@ def get_all_free_blocks(self) -> List[KVCacheBlock]:
 
 
 def hash_block_tokens(parent_block_hash: Optional[int],
-                      curr_block_token_ids: Tuple[int]) -> BlockHashType:
+                      curr_block_token_ids: List[int]) -> BlockHashType:
```

Comment:
It's actually the revers AFAICT - in hash_request_tokens() we were converting the token IDs from their natural list[int] form to tuple[int]

```
block_token_ids = tuple(token_ids[start:end])
```

I think just removing that conversion to tuple makes more sense?

Reply:
ok, to avoid the conversion from ConstantList to list elsewhere, I've added:

```
TokenIdsType = Union[List[int], ConstantList[int]]
BlockHashType = Tuple[int, TokenIdsType]
```

Collection didn't work, because it needs to be indexable/sliceable:

```
    for start in range(0, len(token_ids), block_size):
        end = start + block_size
	block_token_ids = token_ids[start:end]
```

Reply:
How about `Sequence[int]` then?

Reply:
Good idea, done now :+1: